### PR TITLE
Layers -14 Additional List and create layers tests

### PR DESCRIPTION
**Pull request by bhanvimenghani**
## Description

Please describe the issue or feature and the summary of changes made to fix this. 

Fixes # (issue)

### Type of change

- [ ] Bug fix
- [ ] New feature
- [ ] Docs update
- [ ] Breaking change (What changes might users need to make in their application due to this PR?)
- [ ] Requires DB changes

## How has this been tested?

Please describe the tests that were run to verify your changes and steps to reproduce. Please specify any test configuration required. 

- [ ] New Test X
- [ ] Functional testsuite

**Test Configuration**
* Kubernetes clusters tested on: 

## Checklist :dart:

- [ ] Followed coding guidelines
- [ ] Comments added
- [ ] Dependent changes merged
- [ ] Documentation updated
- [ ] Tests added or updated

## Additional information

Include any additional information such as links, test results, screenshots here

## Summary by Sourcery

Expand and harden test coverage for createLayer and listLayers REST APIs, including additional validation scenarios and test utilities.

New Features:
- Add positive tests to ensure createLayer accepts requests when optional fields like details are missing or null.
- Introduce case-sensitivity and creation-order behavior tests for the listLayers API.

Enhancements:
- Broaden negative test coverage for createLayer to cover query/label/tunable validation, JSON parsing, and API structure edge cases.
- Relax test expectations to allow substring matching for dynamic validation error messages.
- Switch createLayer tests to use pytest tmp_path for temporary JSON payload files and add a helper to clean up all layers between tests.
- Document the expanded createLayer and listLayers test scenarios in the local monitoring test guide.

Tests:
- Add new negative tests for invalid query and label definitions, tunable configurations, and malformed or empty request bodies in the createLayer API.
- Add tests for invalid query parameters, special/unicode characters in names, and behavior when no layers exist for the listLayers API.
- Add helper-based cleanup tests to ensure listLayers behavior is validated from a clean state.

### CODE REVIEW

**Comment by kusumachalasani** [comment]
@bhanvimenghani Can you please trigger the jenkins test and post the results ?

File: tests/scripts/local_monitoring_tests/rest_apis/test_list_layers.py [hunk 1]
```diff
@@ -365,3 +389,216 @@ def test_list_layer_with_special_characters_in_name(cluster_type, special_char_n
...
+
+# =============================================================================
+# POSITIVE TEST CASES - Performance and Pagination Tests
+# =============================================================================
+
+@pytest.mark.layers
+@pytest.mark.sanity
+def test_list_layers_performance_with_many_layers(cluster_type):
+    """
+    Test Description: This test validates listLayers API performance when listing 100+ layers.
```
Comment (chandrams) [inline tests/scripts/local_monitoring_tests/rest_apis/test_list_layers.py]: This test should not be part of functional tests. You can add it to load testing and capture the relevant metrics

**Comment by chandrams** [comment]
@bhanvimenghani Can you please address sourcery comments and share the test results

**Comment by chandrams** [comment]
@bhanvimenghani Can you please address sourcery comments and share the test results

File: tests/scripts/local_monitoring_tests/rest_apis/test_layer_detection.py [hunk 2]
```diff
@@ -0,0 +1,331 @@
...
+# =============================================================================
+
+@pytest.mark.layers
+@pytest.mark.detection
+@pytest.mark.parametrize("layer_file", [
+    pytest.param("semeru-actuator-config.json", id="semeru_multiple_queries"),
+    pytest.param("hotspot-micrometer-config.json", id="hotspot_multiple_queries"),
+    pytest.param("quarkus-micrometer-config.json", id="quarkus_single_query"),
+])
+def test_layer_detection_with_valid_query(cluster_type, layer_file):
```
[the remark this answers is not in the document]
Comment (bhanvimenghani) [reply tests/scripts/local_monitoring_tests/rest_apis/test_layer_detection.py]: outdated comment

File: tests/scripts/local_monitoring_tests/rest_apis/test_list_layers.py (hunk 1 above)
Comment (bhanvimenghani) [reply tests/scripts/local_monitoring_tests/rest_apis/test_list_layers.py]: removed the tests from other functional tests.

File: tests/scripts/local_monitoring_tests/rest_apis/test_list_layers.py [hunk 3]
```diff
@@ -365,3 +372,209 @@ def test_list_layer_with_special_characters_in_name(cluster_type, special_char_n
...
+
+
+
+# =============================================================================
+# POSITIVE TEST CASES - Performance and Pagination Tests
+# =============================================================================
+
+@pytest.mark.layers
+@pytest.mark.sanity
+def test_list_layers_performance_with_many_layers(cluster_type):
```
[the remark this answers is not in the document]
Comment (bhanvimenghani) [reply tests/scripts/local_monitoring_tests/rest_apis/test_list_layers.py]: this test is removed now

**Comment by bhanvimenghani** [comment]
[test_create_layer_output.txt](https://github.com/user-attachments/files/27951178/test_create_layer_output.txt)
[test_list_layers_output.txt](https://github.com/user-attachments/files/27951180/test_list_layers_output.txt)
 Locally tests the list and create layer tests, with the image quay.io/bmenghan/layer-demo-e2e:v21 will be triggering the jenkins tests

**Comment by chandrams** [comment]
@sourcery-ai review

**Comment by bhanvimenghani** [comment]
Sourcery comments accommodated , comment from previous pr accommodated , test doc updated

**Comment by bhanvimenghani** [comment]
@sourcery-ai review

File: tests/scripts/local_monitoring_tests/rest_apis/test_create_layer.py [hunk 4]
```diff
@@ -160,6 +161,58 @@ def test_create_layer_with_minimum_required_fields(cluster_type):
...
+    ("without_details", "recommender.com/v1", "KruizeLayer", "test-meta", "test-layer", "SKIP_DETAILS", '{"presence": "always"}', '[{"name": "t1", "value_type": "double", "upper_bound": "100", "lower_bound": "10", "step": 1}]'),
+    ("null_details", "recommender.com/v1", "KruizeLayer", "test-meta", "test-layer", "NULL_DETAILS", '{"presence": "always"}', '[{"name": "t1", "value_type": "double", "upper_bound": "100", "lower_bound": "10", "step": 1}]'),
+    ("empty_details", "recommender.com/v1", "KruizeLayer", "test-meta", "test-layer", "", '{"presence": "always"}', '[{"name": "t1", "value_type": "double", "upper_bound": "100", "lower_bound": "10", "step": 1}]'),
+    ("query_without_key", "recommender.com/v1", "KruizeLayer", "test-meta", "test-layer", "test layer", '{"queries": [{"datasource": "prometheus", "query": "up"}]}', '[{"name": "t1", "value_type": "double", "upper_bound": "100", "lower_bound": "10", "step": 1}]'),
+])
+def test_create_layer_optional_fields(test_name, apiVersion, kind, metadata_name, layer_name, details, layer_presence, tunables, cluster_type, tmp_path):
+    """
```
[the remark this answers is not in the document]
Comment (bhanvimenghani) [reply tests/scripts/local_monitoring_tests/rest_apis/test_create_layer.py:173]: added test

**Comment by bhanvimenghani** [comment]
Jenkins Run link https://ci.app-svc-perf.corp.redhat.com/job/ExternalTeams/job/Autotune/job/kruize_functional_tests/551/
Tests on kind locally
[layer_tests_results.txt](https://github.com/user-attachments/files/28092881/layer_tests_results.txt)

**Review by chandrams — APPROVED** [review]
LGTM

**Review by kusumachalasani — APPROVED** [review]
lgtm!
